README.md

```diff
@@ -139,6 +138,38 @@ locally.
 
 ---
 
+## Using the Bao-Project Nix Shell
+
+Nix provides a robust and reproducible environment by encapsulating all necessary dependencies within a controlled environment. This approach effectively eliminates issues associated with version conflicts and missing packages/dependencies, ensuring a seamless and reliable development experience.
+
+We offer a nix-shell environment containing all essential Bao-Project dependencies, allowing developers to effortlessly work across all Bao-Project repositories. The required dependencies will be automatically fetched. If you already have the necessary packages/dependencies installed, Nix will prioritize the ones it fetched, ensuring reproducibility. Once you exit the nix-shell environment, those packages are no longer accessible. The Nix packages are isolated from your system packages, eliminating any risk of corruption.
+
+You can utilize the Bao-Project nix-shell in two ways:  (1) run a standalone command within the Bao-Project nix-shell environment; (2) or open an interactive Bao-Project nix-shell environment.
+
+**Running a Standalone Command**
+
+This command opens a nix-shell, executes the provided user command, and automatically closes the shell.
+
+```bash
+make nix-shell-run cmd="your command"
+```
+
+**Opening a Nix-Shell Environment**
+
+This command opens an interactive nix-shell with all the Bao-Project dependencies. 
+
+```bash
+make nix-shell
```

Comment:
I was thinking we could follow the same logic as for docker. Having a directory named "nix", with a fully separate Makefile. Then, running make commands in that directory would automatically execute it inside the nix-shell. For example, from a project's top directory, running:

```
make -C ci/nix format-check
```

would run the command `make format-check` command in the nix-shell. This is the same logic we use for docker.

In that makefile we can have other targets for running arbitrary commands and starting the interactive shell as you implemented in the ci.mk file.

Reply:
As I mentioned above, I will implement your suggestion. It also makes sense to me.